coderd/templates.go

```diff
@@ -318,12 +318,28 @@ func (api *API) patchTemplateMeta(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	var validErrs []httpapi.Error
+	if req.MaxTTLMillis < 0 {
+		validErrs = append(validErrs, httpapi.Error{Field: "max_ttl_ms", Detail: "Must be a positive integer."})
+	}
+	if req.MinAutostartIntervalMillis < 0 {
+		validErrs = append(validErrs, httpapi.Error{Field: "min_autostart_interval_ms", Detail: "Must be a positive integer."})
+	}
+
+	if len(validErrs) > 0 {
+		httpapi.Write(rw, http.StatusBadRequest, httpapi.Response{
+			Message:     "Invalid request to update template metadata!",
+			Validations: validErrs,
+		})
+		return
+	}
+
 	count := uint32(0)
 	var updated database.Template
 	err := api.Database.InTx(func(s database.Store) error {
 		// Fetch workspace counts
 		workspaceCounts, err := s.GetWorkspaceOwnerCountsByTemplateIDs(r.Context(), []uuid.UUID{template.ID})
-		if errors.Is(err, sql.ErrNoRows) {
+		if xerrors.Is(err, sql.ErrNoRows) {
```

Comment:
Hmm, should we add some linting for this? The actual `xerrors.Is` function is marked deprecated, and we have a lot of mixed usage in the code-base:

```
❯ rg xerrors.Is | wc -l
20

❯ rg '[^x]errors.Is' | wc -l
104
```

(Note: I'm not suggesting we fix the usage in this PR).

Reply:
AFAIK there are Reasons for using that instead of the original; something about stacktraces.

Reply:
Yeah, there was a case for using `Errorf` and `New` at least, to get the stack trace. But the others are deprecated and are likely not updated in case there are any changes to stdlib. Interestingly, I now noticed `Errorf` was also marked deprecated in favor of `fmt.Errorf` even though the latter does not include a stack trace. Weird.

Reply:
Yeah, might be a good topic for dicussion!

Reply:
@mafredri we can add a linter for it easy.

```golang
	m.Match("errors.$f($*_)").
		Report("Use xerrors.$f to provide additional stacktrace information!")
```